src/styles/calendar.scss

```
@@ -0,0 +1,68 @@
+.calendar {
+  width: $calendarWidth;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  
+  &--monthLength-28 {
+    @for $i from 29 through 31 {
+      .calendar__day--#{$i} {
+        display: none;
+      }
+      .calendar__day--#{$i}::before {
+        content: "";
+      }
+    }
+  }
+
+  &--monthLength-29 {
+    @for $i from 30 through 31 {
+      .calendar__day--#{$i} {
+        display: none;
+      }
+      .calendar__day--#{$i}::before {
+        content: "";
+      }
+    }
+  }
+
+  &--monthLength-30 {
+    .calendar__day--31 {
+      display: none;
+    }
+    .calendar__day--31::before {
+      content: "";
+    }
+  }
```

This won't be necessary after some changes :)

---

What if we used a loop that generates these conditions for us:
```suggestion
    @for $i from 28 through 31 {
    &__month-length--#{$i} &__day:nth-child(#{$i}) ~ &__day{
      display: none;
    }
```